test/conversions.jl

```diff
@@ -55,6 +55,13 @@ module TestConversions
     @test df[:b] == b
     @test df[:c] == c
 
+    df = invoke(convert, (Type{DataFrame}, Associative), DataFrame, di)
```

Comment:
I thought about that. This version seemed easy enough, but I have no problem adding a test targeting OrderedDicts.